src/helpers/SpotLightHelper.js

```diff
@@ -67,6 +64,24 @@ class SpotLightHelper extends Object3D {
 		this.light.updateWorldMatrix( true, false );
 		this.light.target.updateWorldMatrix( true, false );
 
+		// update the local matrix based on the parent and light target transforms
+		if ( this.parent ) {
+
+			this.matrix
+				.copy( this.parent.matrixWorld )
+				.invert()
+				.multiply( this.light.matrixWorld );
+
+		} else {
+
+			this.matrix
+				.copy( this.light.matrixWorld );
+
+		}
+
+		this.matrix.decompose( this.position, this.quaternion, this.scale );
```

Comment:
Why are you doing this -- instead of retaining `this.matrixAutoUpdate = false` ?

Reply:
I suppose I could do that. It would probably be best to se `matrixWorldAutoUpdate` to false but tbh the matrix / matrixWorld update behavior is fairly confusing to me, now. For example the `Object3D.lookAt` function [will call updateWorldMatrix](https://github.com/mrdoob/three.js/blob/a1be53d671b7161ad8479ab47c72de97e34faa68/src/core/Object3D.js#L284) which will automatically update the objects matrixWorld even if `matrixWorldAutoUpdate` is false.

It's not clear to me when they do and don't apply so I fallback to keeping all the values in sync which I think is more clear, anyway.

Reply:
I'd avoid unnecessarily calling `matrix.decompose()` for this helper. JMO, of course.